statsmodels/regression/recursive_ls.py

```diff
@@ -0,0 +1,695 @@
+    statsmodels.tsa.statespace.mlemodel.MLEResults
+    """
+
+    def __init__(self, model, params, filter_results, cov_type='opg',
+                 **kwargs):
+        super(RecursiveLSResults, self).__init__(
+            model, params, filter_results, cov_type, **kwargs)
+
+        self.df_resid = np.inf  # attribute required for wald tests
+
+        # Save _init_kwds
+        self._init_kwds = self.model._get_init_kwds()
+
+        # Save the model specification
+        self.specification = Bunch(**{
+            'k_exog': self.model.k_exog})
+
+    @property
+    def recursive_coefficients(self):
+        """
+        Estimates of regression coefficients, recursively estimated
+
+        Returns
+        -------
+        out: Bunch
+            Has the following attributes:
+
+            - `filtered`: a time series array with the filtered estimate of
+                          the component
+            - `filtered_cov`: a time series array with the filtered estimate of
+                          the variance/covariance of the component
+            - `smoothed`: a time series array with the smoothed estimate of
+                          the component
+            - `smoothed_cov`: a time series array with the smoothed estimate of
+                          the variance/covariance of the component
+            - `offset`: an integer giving the offset in the state vector where
+                        this component begins
+        """
+        out = None
+        spec = self.specification
+        start = offset = 0
+        end = offset + spec.k_exog
+        out = Bunch(
+            filtered=self.filtered_state[start:end],
+            filtered_cov=self.filtered_state_cov[start:end, start:end],
+            smoothed=None, smoothed_cov=None,
+            offset=offset
+        )
+        if self.smoothed_state is not None:
+            out.smoothed = self.smoothed_state[start:end]
+        if self.smoothed_state_cov is not None:
+            out.smoothed_cov = (
+                self.smoothed_state_cov[start:end, start:end])
+        return out
+
+    @cache_readonly
+    def recursive_resid(self):
+        """
+        Recursive residuals
+
+        Returns
+        -------
+        recursive_resid : array_like
+            An array of length `nobs` holding the recursive
+            residuals.
+
+        Notes
+        -----
+        The first `k_exog` residuals are typically unreliable due to
+        initialization.
+
+        """
+        # See Harvey (1989) section 5.4; he defines the standardized
+        # innovations in 5.4.1, but they have non-unit variance, whereas
+        # the standardized forecast errors assume unit variance. To convert
+        # to Harvey's definition, we need to multiply by the standard
+        # deviation.
+        return (self.filter_results.standardized_forecasts_error.squeeze() *
+                self.filter_results.obs_cov[0, 0]**0.5)
+
+    @cache_readonly
+    def cusum(self):
```

Comment:
I think `cusum` is fine. Currently, `resid_xxx` always refer to a single observation.